runtime/metricsview/executor_rewrite_approx_comparisons.go

```diff
@@ -92,6 +99,19 @@ func (e *Executor) rewriteApproxComparisonNode(a *AST, n *SelectNode) bool {
 		n.FromSelect.OrderBy = order
 		n.FromSelect.Limit = a.Root.Limit
 		n.FromSelect.Offset = a.Root.Offset
+
+		// ---- CTE Optimization ---- //
+		// make FromSelect a CTE and set IsCTE flag on FromSelect
+		n.FromSelect.IsCTE = true
+		cte := n.FromSelect
+		a.CTEs = append(a.CTEs, cte)
```

Comment:
In theory, other rewriters might also add to the CTEs, and the same node could end up being added to `a.CTEs` multiple times. To protect agains that, maybe add a something like a `ast.convertToCTE(n *SelectNode)` util func that sets `IsCTE` and only adds to `a.CTEs` if `IsCTE` was false.